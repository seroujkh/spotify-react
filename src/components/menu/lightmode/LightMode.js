import React, { useEffect, useState, useContext } from "react";
import AuthContext from "../../../store/auth-context";
import PropTypes from 'prop-types';

const LightMode = (props) => {

    const [lightMode, setMode] = useState(props.mode);
    const ctx = useContext(AuthContext);


    useEffect(() => {
        setMode(props.mode);
    }, [props.mode]);

    const toggleLightModeHandler = () => {
        if (lightMode === 'light') {
            setMode("dark");
            ctx.setLightMode("dark");
        } else {
            ctx.setLightMode("light");
            setMode("light");
        }
    }

    return (
        <div className="lightmode d-flex justify-content-between px-3 mt-4" onClick={toggleLightModeHandler}>
            <span className="text-white">Light</span>
            <div className="toggle_light ">
                <div className={(lightMode === 'dark' ? ' right ' : ' left ')}></div>
            </div>
            <span className="text-white">Dark</span>
            <div className="active d-none"  data-testid="active">{lightMode}</div>
        </div>
    )
}

LightMode.propTypes = {
    mode: PropTypes.string.isRequired
}
export default LightMode;