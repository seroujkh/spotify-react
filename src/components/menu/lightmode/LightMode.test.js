import { shallow, mount } from 'enzyme';
import React from 'react';
import LightMode, { LightMode as OriginalLightModeClass } from './LightMode';
import { render, screen } from '@testing-library/react';

let T_CURRENTMODE = "light";

describe('LightMode', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<LightMode mode={T_CURRENTMODE} />));

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    // it('changes lightmode by props', () => {
    //     const { rerender } = render(<LightMode mode="dark" />)
    //     // wrapper.setProps({ mode: 'dark' });
    //     expect(screen.getByTestId('active')).toHaveTextContent('dark');
    // });


    it('changes lightmode by click', () => {
        let currentLightMode = T_CURRENTMODE;
        let nextLightMode = currentLightMode == "light" ? "dark" : "light";
        expect(wrapper.find('.active').text()).toBe(currentLightMode);
        wrapper.find('.lightmode').simulate('click');
        expect(wrapper.find('.active').text()).toBe(nextLightMode);
    });



    // describe('toggleLightModeHandler', () => {
    //     let wrapper;

    //     beforeEach(() => wrapper = shallow(<LightMode mode={T_CURRENTMODE} />));


    //     it('updates displayValue', () => {
    //         wrapper.instance().toggleLightModeHandler();
    //         expect(wrapper.state('mode')).toEqual('5');
    //     });
    // });

});