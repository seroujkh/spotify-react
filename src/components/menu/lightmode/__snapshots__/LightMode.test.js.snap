// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightMode should render correctly 1`] = `ShallowWrapper {}`;
